File: src/main/java/io/github/takenoha/towerdefense/runtime/TowerBlockKey.java
package io.github.takenoha.towerdefense.runtime;

import java.util.Objects;
import java.util.UUID;

/** Exact world/block coordinate used to prevent two towers sharing one physical position. */
public record TowerBlockKey(UUID worldId, int x, int y, int z) {
    public TowerBlockKey {
        Objects.requireNonNull(worldId, "worldId");
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/runtime/TowerBlockKey.kt
package io.github.takenoha.towerdefense.runtime

import java.util.Objects
import java.util.UUID
import kotlin.jvm.JvmRecord

/** Exact world/block coordinate used to prevent two towers sharing one physical position. */
@JvmRecord
data class TowerBlockKey(
    val worldId: UUID,
    val x: Int,
    val y: Int,
    val z: Int,
) {
    init {
        Objects.requireNonNull(worldId, "worldId")
    }
}
